rs/interfaces/adapter_client/src/lib.rs

```diff
@@ -2,6 +2,15 @@ use std::time::Duration;
 use strum_macros::IntoStaticStr;
 use thiserror::Error;
 
+/// The maximal number of https outcalls on a single subnet.
+///
+/// This creates a throughput bottleneck, which is required to reason about the maximal
+/// number of consensus artifacts (http outcall shares):
+///
+/// NUMBER_OF_NODES * MAX_CANISTER_HTTP_RESPONSES_IN_FLIGHT * C for some constant C which is
+/// implementation dependent and needs to be estimated manually.
+pub const MAX_CANISTER_HTTP_RESPONSES_IN_FLIGHT: usize = 100;
```

Comment:
I think this is too low. See the comment on the old limit in the adapter client

> To support 100 req/s with a worst case request latency of 30s the queue size needs buffer 100 req/s * 30s = 3000 req.
 The worst case request latency used here should be equivalent to the request timeout in the adapter.
 
 In the worst case (malicious canister) can do requests that take 30s (the timout we use when fetching a url, attacker could implement a super slow endpoint). In that case the throughput of the feature is 3.33 req/s (100 req / 30s). This is quite low since the feature currently (according to benchmarks) can support 50-100 req/s.

Reply:
Let's say we keep it at 3k. Would be comfortable with at least 3k * 28 artifacts on the fiduciary subnet?

Reply:
For p2p this would be okay. Reducing the in-flight request too much is also risky because of the reason I described.

Reply:
Thanks Tim! 

@eichhorl do you have any concerns with using the constant `3000` for the number of requests in-flight wrt pool bounds? These shares are small but not tiny either (each of them contains the 256 bits long hash of the response (additionally to 3 other small fields (wrapping u64s)).